M2/Macaulay2/packages/SuperLinearAlgebra.m2

```diff
@@ -0,0 +1,758 @@
+assert(G.targetM1 == 3)
+assert(G.targetM3 == 2)
+assert(G.sourceM1 == 2)
+assert(G.sourceM2 == 2)
+///
+
+--------------------
+-- parity  
+-------------------- 
+
+parity = method();
+parity (RingElement, Ring, List) := (f, R, oddNumberList) -> (
+    e := symbol e;
+    e = exponents f;
+    l := symbol l;
+    l = {};
+    for i from 0 to (#gens R-1) do (
+        for j from 0 to #oddNumberList-1 do (
+            if R_(i) == oddNumberList_(j) then (
+                l = join(l,{i})
+            )
+        )
+    );
+    d := symbol d;
+    countEvenNumber := symbol countEvenNumber; 
+    
+    d = 0; 
+    countEvenNumber = 0;
+    for i from 0 to (#e-1) do (
+        if (d%2) == 0 then
+             countEvenNumber = countEvenNumber+1;
+        d = 0;
+        for j from 0 to #l-1 do (
+            if (e_i)_(l_j) == 1 then (
+                d = d+1
+            )
+        )
+    );
+    d = 0;
+    for j from 0 to #l-1 do (
+        if (e_(#e-1))_(l_j) == 1 then (
+            d = d+1
+        )
+    ); 
+    
+    if (d%2) == 0 then countEvenNumber = countEvenNumber+1; 
+
+    if (countEvenNumber-1) == #e then 0 
+    else if (countEvenNumber-1) == 0 then 1 
+    else-1
+) 
+
+parity (Number, Ring, List) := (f, R, oddNumberList) -> (
+    0
+) 
+
+TEST ///
+R1 = QQ[x_0..x_3];
+R2 = QQ[z_0, z_1];
+R = superRing(R1, R2);
+a = {z_0, z_1} ;
+g = x_1*x_2*x_3+4;
+f = x_1*x_2*x_3+x_1*z_0+z_1*z_0-4*x_2*z_1*z_0+4;
+h = z_0+z_0*x_0+z_1;
+assert(parity(f, R, a) ==-1)
+assert(parity(g, R, a) == 0)
+assert(parity(h, R, a) == 1)
+assert(parity(1+2.5*ii, R, a) == 0)
+///
+
+--------------------
+-- superMatrixParity
+--------------------
+
+superMatrixParity = method();
+superMatrixParity(SuperMatrix, Ring, List) := (SM, R1, a) -> (
+    m1 := symbol m1; m1 = 0;
+    m2 := symbol m2;
+    m2 = 0;
+    m3 := symbol m3;
+    m3 = 0;
+    m4 := symbol m4;
+    m4 = 0;
+    r1 := symbol r1;
+    r1=SM.targetM1;
+    r2 := symbol r2;
+    r2=SM.targetM3;
+    c1 := symbol c1;
+    c1=SM.sourceM1;
+    c2 := symbol c2;
+    c2=SM.sourceM2;
+    Minor11 := submatrix(SM.supermatrix, {0..(r1-1)}, {0..(c1-1)});
+    Minor22 := submatrix(SM.supermatrix, {r1..(r1+r2-1)}, {c1..(c1+c2-1)});
+    Minor21 := submatrix(SM.supermatrix, {r1..(r1+r2-1)}, {0..(c1-1)});
+    Minor12 := submatrix(SM.supermatrix, {0..(r1-1)}, {c1..(c1+c2-1)});
+    if isSkewCommutative(R1) == true then (
+        fij := symbol fij;
+        count1 :=  symbol count1;
+        count1=0;
+        count11 := symbol count11;
+        count11=0;
+        for i from 0 to (r1-1) do 
+            for j from 0 to (c1-1) do (
+                fij = Minor11_(i, j);
+                if fij == 0 then count1 = count1 
+                else if (parity(fij, R1, a) ==-1) then (count11 = count11 +1) 
+                else if (parity(fij, R1, a) == 1) then count1 = count1+1
+                else if (parity(fij, R1, a) == 0) then count1 = count1
```

Comment:
It's probably inefficient to compute the parity 3 times here.

Reply:
Yes, thanks. You are absolutely right. Even more, one of the conditions `if parity(fij, R1, a) == 0 then ...` was completely additional.